InteractionBase/src/commands/CAddNodeToViewByName.cpp

```diff
@@ -40,18 +41,42 @@ bool CAddNodeToViewByName::canInterpret(Visualization::Item*, Visualization::Ite
 	return commandTokens.size() >= 1 && commandTokens.first() == name();
 }
 
-CommandResult* CAddNodeToViewByName::execute(Visualization::Item*, Visualization::Item* target,
-		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>&)
+CommandResult* CAddNodeToViewByName::execute(Visualization::Item* source, Visualization::Item* target,
+		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& cursor)
 {
 	if (commandTokens.size() < 2)
 		return new CommandResult(new CommandError("Please specify a node to add"));
 
+	auto currentView = target->scene()->currentViewItem();
+	QPoint posToInsert;
+	if (source == currentView && isHorizontal(cursor->region()))
+	{
+		auto below =  findItemBelow(currentView, cursor->region());
```

Comment:
Actually I'm silly. The stuff above only makes sense in cases where the cursor is not directly owned by the view. In your current implementation, you should directly fetch the x, y from the cursor.

It would be nice though to make this command also add nodes closer to the current cursor location, even if the cursor is not directly owned by the view item, in case what I said above about using `focusedElement2DIndex` still makes sense.

Regardless of whether we have a nice behavior also in the case when `source != currentView` you should remove the `findItemLeft` and `findItemBelow`.